lesson15/class_for_rhombus.py

```diff
@@ -0,0 +1,51 @@
+
+class Rhombus:
+    def __init__(self, side_a:int, angle_a_between_sides_a_and_b:int, angle_b_adjacent_to_the_angle_a:int) ->None:
+        self.side_a = side_a
+        self.angle_a_between_sides_a_and_b = angle_a_between_sides_a_and_b
+        self.angle_b_adjacent_to_the_angle_a = angle_b_adjacent_to_the_angle_a
+
+    def __setattr__(self, key, value):
+        if key == 'side_a':
+            if value>0:
+                super().__setattr__(key, value)
+            else:
+                raise ValueError("The side length must be greater than 0.")
+
+        elif key == 'angle_a_between_sides_a_and_b':
```

Comment:
I guess it is better to split this difficult logic of attributes assigning into different methods. 
For ex you will have 3 private methods to setting up attributes for side_a and angles. And in __setattr__ method only call this methods.

Smth like that
`    def __setattr__(self, key, value):
        if key:
            self.__set_atr_if(key, value) `

   ` def __set_atr_if(self, key, value):
        super.__setattr__(self, key, value)`
        
   It will make your code more clear to understand and less complicated.